napari/utils/tree/selection.py

```diff
@@ -0,0 +1,21 @@
+from typing import Iterable, Optional, TypeVar
+
+from ...utils.events import EventedSet
+
+_T = TypeVar("_T")
+
+
+class Selection(EventedSet[_T]):
```

Comment:
Would it be possible to have this both be an `EventedModel` with `current` as its only attribute AND an `EventedSet`. Probably not, but i did want to ask as it would mean we'd get things like the equality check for free.

Reply:
I actually tried that a bit, and it revealed a limitation of the `EventedModel` which is that we can't modify the event emission (for example, here I want the event to have more attributes than just `(value=___)`.  There was that slightly convoluted logic in the `evented_dataclass` that let you override event emission if desired.  I'm not sure we want to bring that back, but it is worth noting that we've lost that.  Need to think about it

Reply:
Been thinking about this a little more this morning.  I'm not sure, at this point, that we necessarily want to use `EventedModel` everywhere we can.  What I _do_ think we should emphasize though, is that every custom type should be useable as a _field_ in an `EventedModel`.  so, basically a field should offer its own `__eq__`, and json support. see #2297

Reply:
> or example, here I want the event to have more attributes than just (value=___)

Is is strictly necessary right now to have the `previous` on the event? I guess the reason why I'm asking is I'm wondering one day if we'll be able to simplify some of our event handling code by automatically wiring up connects and using the `value` property. I guess you could say there's nothing preventing that by having `previous` on there too, but I'm just wondering.

I know we also have more properties on our evented list too.